openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -609,21 +554,8 @@ func (a *managementCPUsOverride) Validate(ctx context.Context, attr admission.At
 
 			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.containers.resources.requests"), c.Resources.Requests, fmt.Sprintf("the pod without workload annotations can not have containers with workload resources %q", resourceName)))
 		}
-	} else {
-		if !doesNamespaceAllowWorkloadType(ns.Annotations, workloadType) { // pod has workload annotation, but the pod does not have workload annotation
-			allErrs = append(allErrs, getPodInvalidWorkloadAnnotationError(pod.Annotations, fmt.Sprintf("the pod can not have workload annotation, when the namespace %q does not allow it", ns.Name)))
-		}
-
-		for _, v := range workloadResourceAnnotations {
-			if len(v) > 1 {
-				allErrs = append(allErrs, field.Invalid(field.NewPath("metadata.annotations"), pod.Annotations, "the pod resource annotation value can not have more than one key"))
-			}
-
-			// the pod should not have any resource annotations with the value that includes keys different from cpushares
-			if _, ok := v[containerResourcesAnnotationValueKeyCPUShares]; len(v) == 1 && !ok {
-				allErrs = append(allErrs, field.Invalid(field.NewPath("metadata.annotations"), pod.Annotations, "the pod resource annotation value should have only cpushares key"))
-			}
-		}
+	} else if !doesNamespaceAllowWorkloadType(ns.Annotations, workloadType) { // pod has workload annotation, but the namespace does not allow specified workload
```

Comment:
You're right this is a bit tough to read, not only with the changes but I think in general the `if else` got a bit much here. I went ahead and updated to use a switch statement and make it easier to reason about. 

Regarding the `else` part, the reason that's there is to not populate the error array with a red herring. If the `workloadType` is empty, we know the other function will fail, since an empty string isn't valid, so might as well avoid adding noise and an error array.

Also, ideally, I think we would avoid having the `doesNamespaceAllowWorkloadType` function handle empty string case, since in my mind calling this function with an empty `workloadType` should fail since the function ins't aware of pod annotations, only namespace annotations.